pkg/webhooks/generation.go

```diff
@@ -85,13 +86,58 @@ func (ws *WebhookServer) HandleGenerate(request *v1beta1.AdmissionRequest, polic
 	return
 }
 
+//HandleUpdateAndDelete handles admission-requests for delete
+func (ws *WebhookServer) handleDelete(request *v1beta1.AdmissionRequest) {
+	resource, err := enginutils.ConvertToUnstructured(request.OldObject.Raw)
+	if err != nil {
+		logger.Error(err, "failed to convert object resource to unstructured format")
+	}
+
+	resLabels := resource.GetLabels()
+	if resLabels["app.kubernetes.io/managed-by"] == "kyverno" && resLabels["generate.kyverno.io/synchronize"] == "enable" {
+		grName := resLabels["generate.kyverno.io/gr-name"]
+		gr, err := ws.grLister.Get(grName)
+		if err != nil {
+			logger.Error(err, "failed to get generate request", "name", grName)
+		}
+		ws.grController.EnqueueGenerateRequestFromWebhook(gr)
+	}
+}
+
+//HandleUpdateAndDelete handles admission-requests for update
+func (ws *WebhookServer) handleUpdate(request *v1beta1.AdmissionRequest) {
+	resource, err := enginutils.ConvertToUnstructured(request.OldObject.Raw)
+	if err != nil {
+		logger.Error(err, "failed to convert object resource to unstructured format")
+	}
+
+	resLabels := resource.GetLabels()
+	if resLabels["generate.kyverno.io/clone-policy-name"] != "" {
+		policyNames := strings.Split(resLabels["generate.kyverno.io/clone-policy-name"], ",")
+		for _, policyName := range policyNames {
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{
+				"generate.kyverno.io/policy-name": policyName,
+			}))
+
+			grList, err := ws.grLister.List(selector)
+			if err != nil {
+				logger.Error(err, "failed to get generate request for the resource", "label", "generate.kyverno.io/policy-name")
+
+			}
+			for _, gr := range grList {
+				ws.grController.EnqueueGenerateRequestFromWebhook(gr)
+			}
+		}
+	}
+}
+
 func (ws *WebhookServer) deleteGR(logger logr.Logger, engineResponse *response.EngineResponse) {
 	logger.V(4).Info("querying all generate requests")
 	selector := labels.SelectorFromSet(labels.Set(map[string]string{
```

Comment:
It would be good to refactor and extract all the label creation / and selection logic into reusable methods